app/controllers/weekly_plans_controller.rb

```diff
@@ -19,26 +17,40 @@ def new
   def create
     time = Benchmark.measure do
       api_client = WeeklyPlanAPIClient.new
+      birth_date = current_user.date_of_birth
+      current_date = Date.today
+      user_age = current_date.year - birth_date.year
+      user_age -= 1 if current_date.month < birth_date.month || (current_date.month == birth_date.month && current_date.day < birth_date.day)
       user_info = {
```

Comment:
Why not passing the user to the service? The hash does a mapping 1to1 of the methods?